drv/lpc55-swd/src/main.rs

```diff
@@ -972,6 +1002,68 @@ impl ServerImpl {
     fn pin_setup(&mut self) {
         setup_pins(self.gpio).unwrap_lite();
     }
+
+    /// Swaps the currently-active SP slot
+    fn swap_sp_slot(&mut self) -> Result<(), Ack> {
```

Comment:
I don't love putting this function in the SWD driver as it seems like the wrong layer but I also don't have a better answer and adding another task has overhead.

Reply:
Same!  Putting it in `lpc55-sprot-server` was definitely a worse place, and I couldn't find a better one.